### PR TITLE
[YUNIKORN-2335] Use go standard library min and max functions

### PR DESCRIPTION
**Pull request by g1geordie**
### What is this PR for?
Use go standard library min and max functions

### What type of PR is it?
* [x] - Refactoring

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2335

### CODE REVIEW

**Comment by chia7712** [comment]
@g1geordie thanks for this patch. it looks great. The following links could use standard function also. please take a look.

1. https://github.com/apache/yunikorn-core/blob/master/pkg/scheduler/objects/preemption.go#L876
2. https://github.com/apache/yunikorn-core/blob/master/pkg/common/resources/resources.go#L453

**Comment by g1geordie** [comment]
@chia7712  Thanks review, I fixed it. Please take a look :)

**Review by pbacsko — DISMISSED** [review]
+1 LGTM

File: pkg/scheduler/objects/preemption.go [hunk 1]
```diff
@@ -727,8 +727,8 @@ func (qps *QueuePreemptionSnapshot) IsWithinGuaranteedResource() bool {
 	if qps.Leaf && guaranteed.IsEmpty() {
 		return false
 	}
-	max := qps.GetMaxResource()
-	absGuaranteed := resources.ComponentWiseMinPermissive(guaranteed, max)
+	maxResource := qps.GetMaxResource()
```
Comment (chia7712) [inline pkg/scheduler/objects/preemption.go:730]: this change is related to IDE warnings ?

File: pkg/scheduler/objects/preemption.go [hunk 2]
```diff
@@ -873,10 +873,7 @@ func preemptPredicateCheck(plugin api.ResourceManagerCallback, ch chan<- *predic
 func batchPreemptionChecks(checks []*si.PreemptionPredicatesArgs, batchSize int) [][]*si.PreemptionPredicatesArgs {
 	var result [][]*si.PreemptionPredicatesArgs
 	for i := 0; i < len(checks); i += batchSize {
-		end := i + batchSize
-		if end > len(checks) {
-			end = len(checks)
-		}
+		end := min(i+batchSize, len(checks))
```
Comment (chia7712) [inline pkg/scheduler/objects/preemption.go:876]: `i + batchSize`

**Review by chia7712 — COMMENTED** [review]
@g1geordie thanks for this patch. two trivial comments below. please take a look.

File: pkg/scheduler/objects/preemption.go (hunk 1 above)
Comment (craigcondit) [reply pkg/scheduler/objects/preemption.go:730]: This change is removing a call to ComponentWiseMinPermissive, which is incorrect.

**Review by craigcondit — CHANGES_REQUESTED** [review]
See comments on preemption changes.

File: pkg/scheduler/objects/preemption.go (hunk 1 above)
Comment (g1geordie) [reply pkg/scheduler/objects/preemption.go:730]: yes, it's IDE warning.
```
Warning:(730, 2) Variable 'max' collides with the 'builtin' function
```

Comment (g1geordie) [reply pkg/scheduler/objects/preemption.go:730]: > This change is removing a call to ComponentWiseMinPermissive, which is incorrect.

I just do a rename by IDE, the call is still like before.

File: pkg/scheduler/objects/preemption.go (hunk 2 above)
Comment (g1geordie) [reply pkg/scheduler/objects/preemption.go:876]: I wanna ask. Is there some format command we can call?
I use Intellij format command to format and also notice it is a littlte different. 
So I try to search `i+` , I see 
image
The format don't have space too, so I ..

Comment (chia7712) [reply pkg/scheduler/objects/preemption.go:876]: it seems there is no strict format :_

**Review by chia7712 — APPROVED** [review]
LGTM

will merge it later

**Comment by craigcondit** [comment]
@chia7712 Do not merge this as-is. It breaks preemption semantics (see previous comments).

File: pkg/scheduler/objects/preemption.go (hunk 1 above)
Comment (craigcondit) [reply pkg/scheduler/objects/preemption.go:730]: Sorry, missed that it was showing up under the comments.

**Review by craigcondit — APPROVED** [review]
+1 LGTM. Sorry about the false alarm on conflicting changes.

**Comment by chia7712** [comment]
> Sorry about the false alarm on conflicting changes.

thanks for confirmation!

**Pull request closed without merging**